util/promlint/promlint.go

```diff
@@ -232,17 +295,41 @@ func metricUnits(m string) (unit string, base string, ok bool) {
 // Units and their possible prefixes recognized by this library.  More can be
 // added over time as needed.
 var (
-	baseUnits = []string{
-		"amperes",
-		"bytes",
-		"candela",
-		"grams",
-		"kelvin", // Both plural and non-plural form allowed.
-		"kelvins",
-		"meters", // Both American and international spelling permitted.
-		"metres",
-		"moles",
-		"seconds",
+	// map a unit to the appropriate base unit.
+	units = map[string]string{
+		// Base units.
+		"amperes": "amperes",
+		"bytes":   "bytes",
+		"candela": "candela",
+		"grams":   "grams",
+		"celsius": "celsius", // Allow both kelvin and celsius.
+		"kelvin":  "kelvin",  // Both plural and non-plural form allowed.
+		"kelvins": "kelvins",
+		"meters":  "meters", // Both American and international spelling permitted.
+		"metres":  "metres",
+		"moles":   "moles",
```

Comment:
Wouldn't the base unit be the number of molecules?

Time and data size are the main ones, I wouldn't worry about the rest too much.

Reply:
https://en.wikipedia.org/wiki/Mole_(unit)

> The mole is the base unit of amount of substance ("number of substance") in the International System of Units or System International (SI) [...]

Reply:
That's basically a prefix, and we don't use the SI unit for mass for example as it has a prefix.

I don't expect many users to be doing chemistry with Prometheus, let's not get beyond what you're likely to see in a standard datacenter.

Reply:
Does it seem reasonable to lint the classes listed in https://prometheus.io/docs/practices/naming/#base-units and delete the rest (in this case moles and candela)?

Reply:
Yes

Reply:
:+1: I removed moles and candela.